src/planscape/climate_foresight/models.py

```diff
@@ -70,6 +70,78 @@ def __str__(self):
         return f"{self.name} - {self.planning_area.name}"
 
 
+class ClimateForesightPillar(CreatedAtMixin, models.Model):
+    """
+    Represents pillars for organizing climate foresight data layers.
+
+    Global pillars (null run_id) represent the 10 pillars of fire resilience and cannot be deleted.
+    Run-specific pillars are custom pillars created by users and can only be deleted
+    when the run is in draft mode.
+    """
+
+    run_id = models.ForeignKey(
+        ClimateForesightRun,
+        on_delete=models.CASCADE,
+        related_name="custom_pillars",
+        null=True,
+        blank=True,
+        db_column="run_id",
+        help_text="If null, this is a global/shared pillar. If set, it's specific to that run.",
+    )
+
+    name = models.CharField(max_length=255, help_text="Name of the pillar")
+
+    order = models.IntegerField(default=0, help_text="Display order for the pillar")
+
+    created_by = models.ForeignKey(
+        User,
+        on_delete=models.CASCADE,
+        related_name="created_climate_foresight_pillars",
+        help_text="User who created this pillar",
+    )
+
+    class Meta:
+        ordering = ["order", "name"]
+        verbose_name = "Climate Foresight Pillar"
+        verbose_name_plural = "Climate Foresight Pillars"
+        constraints = [
+            models.UniqueConstraint(
+                fields=["name"],
+                condition=models.Q(run_id__isnull=True),
+                name="unique_global_climate_foresight_pillar_name",
+            ),
+            models.UniqueConstraint(
+                fields=["run_id", "name"],
+                condition=models.Q(run_id__isnull=False),
+                name="unique_run_climate_foresight_pillar_name",
+            ),
+        ]
+
+    def __str__(self):
+        scope = "Custom" if self.is_custom else "Global"
+        return f"{self.name} ({scope})"
+
+    @property
+    def is_custom(self) -> bool:
+        """Returns True if this is a custom (run-specific) pillar."""
+        return self.run_id is not None
+
+    def can_delete(self) -> bool:
```

Comment:
This by itself does nothing. You can still issue `model.delete()` or do a delete with a queryset (like Foo.objects.filter().delete()`

If you need to prevent deletion, I would at least override the `delete` method to prevent unwanted deletion.